### PR TITLE
adds data tables to more course visualizations 

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes https://github.com/ilios/ilios/issues/5584
fixes https://github.com/ilios/ilios/issues/5583

### CODE REVIEW

**Comment by dartajax** [comment]
image

**Comment by dartajax** [comment]
image

**Comment by dartajax** [comment]
this course was used for the screen shot above ...

http://localhost:4200/data/courses/2224

2224 is the course ID

**Comment by stopfstedt** [comment]
> this course was used for the screen shot above ...
> 
> http://localhost:4200/data/courses/2224
> 
> 2224 is the course ID

@dartajax - not sure where we left it on this particular aspect. render a donut here instead, or not?

**Comment by stopfstedt** [comment]
> this course was used for the screen shot above ...
> 
> http://localhost:4200/data/courses/2224
> 
> 2224 is the course ID

also - the graph is "staying within it's quadrant", but it's bumping up against the next one.

image

i could give it a bit of room though

image

@dartajax - given some extra room - would that cut it for you?

**Comment by dartajax** [comment]
> > this course was used for the screen shot above ...
> > http://localhost:4200/data/courses/2224
> > 2224 is the course ID
> 
> @dartajax - not sure where we left it on this particular aspect. render a donut here instead, or not?

I think I like the bars myself but have put it up for team discussion - it seems like more information is available sooner - and I like that.

**Comment by dartajax** [comment]
> > this course was used for the screen shot above ...
> > http://localhost:4200/data/courses/2224
> > 2224 is the course ID
> 
> also - the graph is "staying within it's quadrant", but it's bumping up against the next one.
> 
> image
> 
> i could give it a bit of room though
> 
> image
> 
> @dartajax - given some extra room - would that cut it for you?

Why not just make it straight quadrants (quarters) of the screen available to each visualization / graph in case we end up converting the Instructors to be bar chart as well? The extra room is helpful though and would cut it for me.

**Comment by dartajax** [comment]
formatting suggestion ...

old ...
image

updated ...
image

**Comment by stopfstedt** [comment]
> formatting suggestion ...

i like this, will give it a try. thanks for the suggestion.

**Review by jrjohnson — APPROVED** [review]
Nice test detail, I didn't validate the data manipulation because my eyes glazed over after the first few, but given the extensive click testing so far and the test coverage I think this is ready. I made a comment about `slice` that doesn't need to be part of this change, more for informational purposes.

File: packages/ilios-common/addon/components/course/visualize-instructor-session-type-graph.js [hunk 1]
```diff
@@ -1,47 +1,77 @@
...
+      prop += ':desc';
     }
-    return this.loadedData;
+    this.sortBy = prop;
   }
 
-  async getData(sessions) {
-    if (!sessions) {
+  async getData(course, user) {
+    const sessions = (await course.sessions).slice();
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply packages/ilios-common/addon/components/course/visualize-instructor-session-type-graph.js]: good catch. general cleanup-ticket and PR incoming after this lands.

corrected heavy_check_mark:

**Review by jrjohnson — APPROVED** [review]

**Comment by saschaben** [comment]
looking good! I think we have arrived for this PR. remaining issues will be follow-ons and up for discussion.

**Comment by dartajax** [comment]
The fonts appear slightly larger than what I am running on my local proxied to a slightly older demo - is that due to this PR or something else? My guess is something else.

**Comment by dartajax** [comment]
seems pretty cool to me - I'm cool with documenting how to export the data to Excel - user will have to input the headers but that is not a huge deal

**Review by dartajax — APPROVED** [review]

**Pull request merged**